async_producer.go

```diff
@@ -218,7 +218,16 @@ func (p *asyncProducer) topicDispatcher() {
 
 		if msg.flags&shutdown != 0 {
```

Comment:
Working through that did give me an idea for an additional minor simplification though (see the third commit).